app/ts/App.tsx

```diff
@@ -29,6 +30,60 @@ const WalletComponent = ({ maybeAccountAddress, loadingAccount, isWindowEthereum
 		</button>
 	)
 }
+
+interface TabsProps {
+	maybeAccountAddress: OptionalSignal<AccountAddress>
+	areContractsDeployed: Signal<boolean | undefined>
+}
+const Tabs = ({ maybeAccountAddress, areContractsDeployed }: TabsProps) => {
+	const activeTab = useSignal(0)
+	const tabs = [
+		{ title: 'Trading', path: 'trading', component: <DeployContract maybeAccountAddress = { maybeAccountAddress } areContractsDeployed = { areContractsDeployed }/> },
+		{ title: 'Market Creation', path: 'market-creation', component: <CreateYesNoMarket maybeAccountAddress = { maybeAccountAddress }/> },
+		{ title: 'Reporting', path: 'reporting', component: <Reporting maybeAccountAddress = { maybeAccountAddress }/> },
+		{ title: 'Claim Funds', path: 'claim-funds', component: <ClaimFunds maybeAccountAddress = { maybeAccountAddress }/> }
+	]
+
+	useEffect(() => {
+		const path = window.location.hash.replace('#/', '')
+		const tabIndex = tabs.findIndex(tab => tab.path === path)
+		if (tabIndex !== -1) {
+			activeTab.value = tabIndex
+		}
+	}, [])
+
+	const handleTabClick = (index: number) => {
+		if (tabs[index] === undefined) throw new Error(`invalid Tab index: ${ index }`)
+		activeTab.value = index
+		window.location.hash = `#/${ tabs[index].path }`
+	}
```

Comment:
World's smallest router, big fan.  Most people bring in a giant routing library just to do this and it is annoying.

That being said, we may want to use `window.history.pushState(...)` for this instead.  See https://developer.mozilla.org/en-US/docs/Web/API/History_API and https://stackoverflow.com/a/44850073 for details.  It is not immediately obviously to me that we want `pushState`, but we should understand the tradeoffs.

Reply:
added issue